nef/Preferences/Data/PreferencesDataSource.swift

```diff
@@ -0,0 +1,68 @@
+//  Copyright © 2019 The nef Authors.
+
+import Foundation
+
+class PreferencesDataSource {
+    
+    let fileManager: FileManager
+    
+    init(fileManager: FileManager) {
+        self.fileManager = fileManager
+        buildFileSystem()
+    }
+    
+    var state: PreferencesModel { retrieveState() ?? self.default }
+    
+    var `default`: PreferencesModel {
+        return PreferencesModel(showLines: true,
+                                showWatermark: true,
+                                font: .firaCode,
+                                theme: .dracula,
+                                size: .x2,
+                                color: .nef)
+    }
+    
+    func persist(model: PreferencesModel) {
```

Comment:
Why not calling `store(state:)` directly?

Reply:
Good catch! yep, we can do directly, I suppose I started to work in another direction and I forget to clean up it.